### ISSUE TITLE
`attr` values in IPTB are not recognized.

### ISSUE THREAD

**Issue by LastExile16** [ipfs/iptb#114]
The proper way to use _attributes_ is not available on the readme page but with referring to #50 @whyrusleeping shows some interesting features of `iptb`. I have tried to test some of these attributes but the `iptb` doesn't recognize them.
```
$ iptb attr set 0 latency 50ms
no attribute to set
$ iptb attr set 1 latency 50ms
no attribute to set
$ iptb attr get 1 latency
unrecognized attribute: latency
github.com/ipfs/iptb-plugins.GetAttr
	~/go/src/github.com/ipfs/iptb-plugins/util.go:46
github.com/ipfs/iptb-plugins/local.(*LocalIpfs).Attr
	~/go/src/github.com/ipfs/iptb-plugins/local/localipfs.go:432
gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands.glob..func2
	~/go/src/gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands/attr.go:118
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.HandleAction
	~/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:485
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run
	~/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:193
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).RunAsSubcommand
	~/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:374
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.startApp
	~/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:280
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run
	~/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:79
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).Run
	~/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:250
main.main
	~/go/src/github.com/ipfs/iptb-plugins/iptb/iptb.go:70
runtime.main
	/usr/local/go/src/runtime/proc.go:201
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1333
```

**Comment by Stebalien** [comment]
Well, we need to fix that panic. However, to use latency etc. you'll need to use the *docker* plugin, not the localipfs plugin.

**Comment by Stebalien** [comment]
Actually, that's not a panic. It's just a stack attached to the error message. That shouldn't print out by default, did you modify something?

**Comment by Stebalien** [comment]
*That's* a bug. Unfortunately, *correctly* fixing this will require a bit of work. We'll probably want to switch to using github.com/vishvananda/netlink to manage link state.

**Comment by Stebalien** [comment]
Note: You should still be able to *set* the latency. However, you currently have to *start* your nodes first. Ideally, you wouldn't have to do that (also a bug).

Bug: https://github.com/ipfs/iptb-plugins/issues/19

**Comment by Stebalien** [comment]
Closing as those two bugs should cover everything in this issue.

(please continue filing issues as you find them :smile:)

**Comment by LastExile16** [comment]
I don't know if I have to open a new issue or not, but I'm still not able to run these commands properly, am I doing something wrong!?
It's the same for `bandwidth` and `jitter`
```
$ iptb testbed create --count 5 -type dockeripfs
testbed nodes already exist, overwrite? [y/n]
y
$ iptb init
$ iptb start
$ iptb attr get 0 latency
unrecognized attribute: latency
github.com/ipfs/iptb-plugins.GetAttr
	/sehome/home/nawras/go/src/github.com/ipfs/iptb-plugins/util.go:46
github.com/ipfs/iptb-plugins/docker.(*DockerIpfs).Attr
	/sehome/home/nawras/go/src/github.com/ipfs/iptb-plugins/docker/dockeripfs.go:429
gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands.glob..func2
	/sehome/home/nawras/go/src/gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands/attr.go:118
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.HandleAction
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:485
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:193
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).RunAsSubcommand
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:374
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.startApp
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:280
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:79
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).Run
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:250
main.main
	/sehome/home/nawras/go/src/github.com/ipfs/iptb-plugins/iptb/iptb.go:70
runtime.main
	/usr/local/go/src/runtime/proc.go:201
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1333
$ iptb attr set 0 latency 50ms
container-side interface not found

```

**Comment by LastExile16** [comment]
Further looking for the problem, I noticed the docker container cannot be started by `iptb start` command


```
$ iptb testbed create --count 5 --type  dockeripfs
$ iptb init
$ iptb start
```


--- checking the docker
```
$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
ipfs/go-ipfs        latest              2f5ded240628        2 months ago        43.2MB


$ docker ps -a
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                     PORTS               NAMES
d9baa9b89087        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   10 seconds ago      Exited (1) 8 seconds ago                       serene_bardeen
4b0203b6cb08        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   10 seconds ago      Exited (1) 7 seconds ago                       frosty_chatterjee
b614baeaecf2        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   10 seconds ago      Exited (1) 8 seconds ago                       zen_jones
30e7df626c34        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   10 seconds ago      Exited (1) 8 seconds ago                       stoic_jennings
45b44abf175f        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   10 seconds ago      Exited (1) 8 seconds ago                       inspiring_nobel
```


--- back to iptb

```
$ iptb shell 0
Error: No such container: Unable to find image 'ipfs/go-ipfs:latest' locally
latest: Pulling from ipfs/go-ipfs
0de338cf4258: Pulling fs layer
11c62d923848: Pulling fs layer
365846b6db74: Pulling fs layer
068806c0c7c7: Pulling fs layer
c974b7720ca1: Pulling fs layer
586d5e183876: Pulling fs layer
2851d4256382: Pulling fs layer
9b9ad1445fab: Pulling fs layer
9b9ad1445fab: Waiting
068806c0c7c7: Waiting
c974b7720ca1: Waiting
586d5e183876: Waiting
2851d4256382: Waiting
365846b6db74: Download complete
0de338cf4258: Download complete
0de338cf4258: Pull complete
068806c0c7c7: Verifying Checksum
068806c0c7c7: Download complete
c974b7720ca1: Verifying Checksum
c974b7720ca1: Download complete
11c62d923848: Download complete
586d5e183876: Verifying Checksum
586d5e183876: Download complete
11c62d923848: Pull complete
365846b6db74: Pull complete
068806c0c7c7: Pull complete
9b9ad1445fab: Verifying Checksum
9b9ad1445fab: Download complete
c974b7720ca1: Pull complete
586d5e183876: Pull complete
2851d4256382: Verifying Checksum
2851d4256382: Download complete
2851d4256382: Pull complete
9b9ad1445fab: Pull complete
Digest: sha256:4901bf2adb79a10fa420454a67dfd2417418fac517fead723b085cd4a80e6dfc
Status: Image is up to date for ipfs/go-ipfs:latest
b614baeaecf2e905880bc5acd1cb01981165bea24c67123bec6934e3130aae5d
exit status 1
```

repeating...
```
$ iptb shell 0
# same output as above, it looks like it doesn't recognize the container or image
```


Trying a different command

```
$ iptb attr get 0 bandwidth
unrecognized attribute: bandwidth
github.com/ipfs/iptb-plugins.GetAttr
	/sehome/home/nawras/go/src/github.com/ipfs/iptb-plugins/util.go:46
github.com/ipfs/iptb-plugins/docker.(*DockerIpfs).Attr
	/sehome/home/nawras/go/src/github.com/ipfs/iptb-plugins/docker/dockeripfs.go:429
gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands.glob..func2
	/sehome/home/nawras/go/src/gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands/attr.go:118
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.HandleAction
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:485
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:193
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).RunAsSubcommand
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:374
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.startApp
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:280
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:79
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).Run
	/sehome/home/nawras/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:250
main.main
	/sehome/home/nawras/go/src/github.com/ipfs/iptb-plugins/iptb/iptb.go:70
runtime.main
	/usr/local/go/src/runtime/proc.go:201
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1333
```



Running `iptb start` again, will create new containers for the nodes

```
$ iptb start
```


--- docker output

```
$ docker ps -as
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                     PORTS               NAMES               SIZE
cf9295860263        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 seconds ago       Exited (1) 3 seconds ago                       wonderful_carson    0B (virtual 43.2MB)
8653c3fc6e09        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 seconds ago       Exited (1) 3 seconds ago                       admiring_shockley   0B (virtual 43.2MB)
302b9dbb2672        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 seconds ago       Exited (1) 3 seconds ago                       lucid_mclaren       0B (virtual 43.2MB)
62bb00a20714        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 seconds ago       Exited (1) 3 seconds ago                       dazzling_keldysh    0B (virtual 43.2MB)
686434204a18        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 seconds ago       Exited (1) 3 seconds ago                       stupefied_lumiere   0B (virtual 43.2MB)
d9baa9b89087        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 minutes ago       Exited (1) 5 minutes ago                       serene_bardeen      0B (virtual 43.2MB)
4b0203b6cb08        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 minutes ago       Exited (1) 5 minutes ago                       frosty_chatterjee   0B (virtual 43.2MB)
b614baeaecf2        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 minutes ago       Exited (1) 5 minutes ago                       zen_jones           0B (virtual 43.2MB)
30e7df626c34        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 minutes ago       Exited (1) 5 minutes ago                       stoic_jennings      0B (virtual 43.2MB)
45b44abf175f        ipfs/go-ipfs        "/sbin/tini -- /usr/…"   5 minutes ago       Exited (1) 5 minutes ago                       inspiring_nobel     0B (virtual 43.2MB)
```


--- back to iptb

```
$ iptb stop
exit status 1: Error response from daemon: Cannot kill container: 8653c3fc6e09144bbb2e2c475fe9502b23b13cc56ccfa92c1a9eda5dd59d7925: Container 8653c3fc6e09144bbb2e2c475fe9502b23b13cc56ccfa92c1a9eda5dd59d7925 is not running

node[0]
gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands.mapWithOutput.func1
	/sehome/home/nawras/go/src/gx/ipfs/QmckeQ2zrYLAXoSHYTGn5BDdb22BqbUoHEHm8KZ9YWRxd1/iptb/commands/utils.go:195
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1333
```


***Only after the second `iptb start` command, which means after creating another list of containers, then the `iptb shell ` looks like identifies the container but was not able to start it

```
iptb shell 0
Error response from daemon: Container 8653c3fc6e09144bbb2e2c475fe9502b23b13cc56ccfa92c1a9eda5dd59d7925 is not running
exit status 1
```



--- docker, when trying to start manually

```
$ docker start -a 8653c3fc6e09
Changing user to ipfs
chown: /data/ipfs/version: Operation not permitted
chown: /data/ipfs/config: Operation not permitted
chown: /data/ipfs/datastore/CURRENT: Operation not permitted
chown: /data/ipfs/datastore/000003.log: Operation not permitted
chown: /data/ipfs/datastore/MANIFEST-000004: Operation not permitted
chown: /data/ipfs/datastore/LOCK: Operation not permitted
chown: /data/ipfs/datastore/LOG: Operation not permitted
chown: /data/ipfs/datastore/000002.ldb: Operation not permitted
chown: /data/ipfs/datastore: Operation not permitted
chown: /data/ipfs/datastore: Operation not permitted
chown: /data/ipfs/keystore: Permission denied
chown: /data/ipfs/datastore_spec: Operation not permitted
chown: /data/ipfs/dockerid: Operation not permitted
chown: /data/ipfs/blocks/R3/CIQBED3K6YA5I3QQWLJOCHWXDRK5EXZQILBCKAPEDUJENZ5B5HJ5R3A.data: Operation not permitted
chown: /data/ipfs/blocks/R3: Operation not permitted
chown: /data/ipfs/blocks/R3: Operation not permitted
chown: /data/ipfs/blocks/X3/CIQFTFEEHEDF6KLBT32BFAGLXEZL4UWFNWM4LFTLMXQBCERZ6CMLX3Y.data: Operation not permitted
chown: /data/ipfs/blocks/X3: Operation not permitted
chown: /data/ipfs/blocks/X3: Operation not permitted
chown: /data/ipfs/blocks/OO/CIQBT4N7PS5IZ5IG2ZOUGKFK27IE33WKGJNDW2TY3LSBNQ34R6OVOOQ.data: Operation not permitted
chown: /data/ipfs/blocks/OO: Operation not permitted
chown: /data/ipfs/blocks/OO: Operation not permitted
chown: /data/ipfs/blocks/IL/CIQJFGRQHQ45VCQLM7AJNF2GF5UHUAGGHC6LLAH6VYDEKLQMD4QLILY.data: Operation not permitted
chown: /data/ipfs/blocks/IL: Operation not permitted
chown: /data/ipfs/blocks/IL: Operation not permitted
chown: /data/ipfs/blocks/SHARDING: Operation not permitted
chown: /data/ipfs/blocks/PM/CIQKNNRB2NFYXUZDJ2UWNMSKYLGTKUYDRQTJCDI7JTUDFH6YOYNUPMA.data: Operation not permitted
chown: /data/ipfs/blocks/PM: Operation not permitted
chown: /data/ipfs/blocks/PM: Operation not permitted
chown: /data/ipfs/blocks/diskUsage.cache: Operation not permitted
chown: /data/ipfs/blocks/QV/CIQOHMGEIKMPYHAUTL57JSEZN64SIJ5OIHSGJG4TJSSJLGI3PBJLQVI.data: Operation not permitted
chown: /data/ipfs/blocks/QV: Operation not permitted
chown: /data/ipfs/blocks/QV: Operation not permitted
chown: /data/ipfs/blocks/7J/CIQKKLBWAIBQZOIS5X7E32LQAL6236OUKZTMHPQSFIXPWXNZHQOV7JQ.data: Operation not permitted
chown: /data/ipfs/blocks/7J: Operation not permitted
chown: /data/ipfs/blocks/7J: Operation not permitted
chown: /data/ipfs/blocks/_README: Operation not permitted
chown: /data/ipfs/blocks/B5/CIQDOZU3EAGXWK3PLVFOFOZOAE5USX3XM6I5CSHSQGTML2BAGN7MB5I.data: Operation not permitted
chown: /data/ipfs/blocks/B5: Operation not permitted
chown: /data/ipfs/blocks/B5: Operation not permitted
chown: /data/ipfs/blocks/JN/CIQPHMHGQLLZXC32FQQW2YVM4KGFORVFJAQYY55VK3WJGLZ2MS4RJNQ.data: Operation not permitted
chown: /data/ipfs/blocks/JN: Operation not permitted
chown: /data/ipfs/blocks/JN: Operation not permitted
chown: /data/ipfs/blocks/L5/CIQBIQXZ4NWWDXUSIYSCX7RE6EBXHMGENZNMUDEMGNKMGT2K6LLUL5Y.data: Operation not permitted
chown: /data/ipfs/blocks/L5: Operation not permitted
chown: /data/ipfs/blocks/L5: Operation not permitted
chown: /data/ipfs/blocks/7R/CIQIVZGLOUFY5L4TEYD5WLSCRDDEAW2TNVZSH3OBM5UKNOREHYSY7RA.data: Operation not permitted
chown: /data/ipfs/blocks/7R: Operation not permitted
chown: /data/ipfs/blocks/7R: Operation not permitted
chown: /data/ipfs/blocks/LG/CIQJBQD2O6K4CGJVCCTJNUP57QHR4SKHZ74OIITBBGLOMCO3ZOLWLGA.data: Operation not permitted
chown: /data/ipfs/blocks/LG: Operation not permitted
chown: /data/ipfs/blocks/LG: Operation not permitted
chown: /data/ipfs/blocks/N2/CIQDWKPBHXLJ3XVELRJZA2SYY7OGCSX6FRSIZS2VQQPVKOA2Z4VXN2I.data: Operation not permitted
chown: /data/ipfs/blocks/N2: Operation not permitted
chown: /data/ipfs/blocks/N2: Operation not permitted
chown: /data/ipfs/blocks/2H/CIQN5PVU4ECEKNEVXWRYKCPTVMIPHP5AWWKOCGCPXVNMF7ZWU6UG2HI.data: Operation not permitted
chown: /data/ipfs/blocks/2H: Operation not permitted
chown: /data/ipfs/blocks/2H: Operation not permitted
chown: /data/ipfs/blocks/SH/CIQL3XIOKVDAW5KQF6NNWGFFYAHEQP63TJOVZHAEO7XZBD7KQOCSSHY.data: Operation not permitted
chown: /data/ipfs/blocks/SH: Operation not permitted
chown: /data/ipfs/blocks/SH: Operation not permitted
chown: /data/ipfs/blocks: Operation not permitted
chown: /data/ipfs/blocks: Operation not permitted
chown: /data/ipfs: Operation not permitted
chown: /data/ipfs: Operation not permitted
```

**Comment by Stebalien** [comment]
Everything related to `iptb attr get` is ipfs/iptb-plugins#18. For everything else, I really don't know.

@travisperson?
